src/blocks/map/components/edit.js

```diff
@@ -299,15 +302,28 @@ class Edit extends Component {
 					>
 						<TextControl
 							className="components-placeholder__input"
-							value={ this.state.address }
+							value={ this.state.addressText }
 							placeholder={ __( 'Search for a place or address…' ) }
-							onChange={ nextAddress => this.setState( { address: nextAddress } ) }
+							onChange={ nextAddress => this.setState( { addressText: nextAddress } ) }
 							onKeyDown={ ( { keyCode } ) => handleKeyDown( keyCode ) }
 						/>
 						<Button isLarge type="button" onClick={ renderMap }>
 							{ __( 'Apply' ) }
 						</Button>
 
+						<Button
+							className="components-placeholder__cancel-button"
+							title={ __( 'Cancel' ) }
+							isLink
+							onClick={ () => {
+								setAttributes( { pinned: ! pinned } );
+								this.setState( { addressText: address } );
```

Comment:
Instead of two properties in state you could change this to `this.setState( { address: this.props.attributes.address } )`

Reply:
It seems so obvious now. Thanks for they eyes JR. I learn something new every time you review something of mine.